svg-2-android-vector/src/main/java/com/quittle/svg2androidvector/Svg2AndroidVectorPlugin.java

```diff
@@ -53,14 +56,25 @@ public void apply(final Project project) {
     }
 
     private static void onAfterEvaluate(final Project project, final Svg2AndroidVectorExtension extension) {
+        // The base Android extension, where all resources can be found.
+        final BaseExtension androidExtension = project.getExtensions().findByType(BaseExtension.class);
+        if (androidExtension == null) {
+            // This might mean it was applied to a non-Android project, possibly just the root project.
+            return;
```

Comment:
Rather than silently returning, can you instead throw a descriptive exception that explains either that either Android plugin wasn't applied or that this plugin should be applied to a child project that does apply the Android plugin?

Reply:
I mean, we could do that, but I don't know if it's useful to do so - it means you can't add it in the plugins DSL of your root project and have it auto-apply to children (I think). Other plugins, e.g. ktlint-gradle, https://github.com/JLLeitschuh/ktlint-gradle say explicitly "The plugin can be applied to any project, but only activates if that project has the kotlin plugin applied.". Lacking other examples, I just assumed that was the idiom, since there are other plugins I have in the my project that I don't have `apply false` after and they still work fine.

The usage that this would error on is this, in the root project:

```
plugins {
    id 'com.quittle.svg-2-android-vector' version '0.0.5'
}
```

which I believe applies it to the root project and all sub-projects


and would presumably suggest the following instead: this in the root project:

```
plugins {
    id 'com.quittle.svg-2-android-vector' version '0.0.5' apply false
}
```

and this in every sub-project:

```
plugins {
    id 'com.quittle.svg-2-android-vector'
}
```


(see https://docs.gradle.org/current/userguide/plugins.html#sec:subprojects_plugins_dsl )

Reply:
OK I added an info log message:

```
> Configure project :
Evaluating root project 'gradle-svg-2-android-vector' using build file 'E:\src-ssd\third-party\gradle-svg-2-android-vector\build.gradle'.
Compiling build file 'E:\src-ssd\third-party\gradle-svg-2-android-vector\build.gradle' using SubsetScriptTransformer.
Invalidating in-memory cache of E:\gradle_user_home\caches\journal-1\file-access.bin
Invalidating in-memory cache of E:\src-ssd\third-party\gradle-svg-2-android-vector\.gradle\buildOutputCleanup\outputFiles.bin
Selected primary task ':jar' from project :
Compiling build file 'E:\src-ssd\third-party\gradle-svg-2-android-vector\build.gradle' using BuildScriptTransformer.
root project 'gradle-svg-2-android-vector' had the Svg2AndroidVector plugin applied, but it's not an Android project (maybe the root project?) so we'll not do anything.

> Configure project :example-android-project
Evaluating project ':example-android-project' using build file 'E:\src-ssd\third-party\gradle-svg-2-android-vector\example-android-project\build.gradle'.
Compiling build file 'E:\src-ssd\third-party\gradle-svg-2-android-vector\example-android-project\build.gradle' using BuildScriptTransformer.
Creating configuration androidTestUtil
Applying Svg2AndroidVector plugin to project ':example-android-project'.
```

Reply:
Ah, I see. This makes sense, I'm not sure what standard is but I do like the idea of it "just working" regardless of how you apply it. I doubt it'll cause confusion if someone is applying to the root project which doesn't have the android plugin applied.